### PR TITLE
Adds support for Chevrolet Bolt EV, Bolt EUV

### LINKED ISSUE #42: Add support for Chevrolet EVs
**Manufacturer**
Chevrolet

**Vehicle Model**
Bolt

**Link to the manufacturer's inventory webpage**
https://www.chevrolet.com/electric/shopping/inventory/search?model=Bolt%20EV&radius=250&year=2022&zipCode=90210

### PR DESCRIPTION
**Pull request by blamejoel**
This change resolves #42 with a starting point for Chevrolet EV support.

### CODE REVIEW

**Comment by Ben-Chapman** [comment]
Amazing! Thanks for taking the time to put this together, I'll take a look at the PR this weekend.

**Review by Ben-Chapman — CHANGES_REQUESTED** [review]
This looks really good, thanks for sending it in.

Just a few minor changes needed, see inline comments. Can you also pull in the `dealerUrl` changes detailed in https://github.com/Ben-Chapman/EVFinder/issues/56

File: api/src/blueprints/chevrolet.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          ]
+  }
+
+  if validate_request(params.items()):
+    # Make a call to the API
+    inventory = requests.post(
+      url=api_url,
+      headers=headers,
+      json=inventory_post_data,
+      verify=False
```
[the remark this answers is not in the document]
Comment (blamejoel) [reply api/src/blueprints/chevrolet.py:61]: Just an artifact from using volkswagen.py as a starting point reference. Removed!

File: app/src/manufacturers/chevroletMappings.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  * }
+  */
+
+const chevroletInventoryMapping = {
+  'intColor': 'interiorColor',
+  'extColor': 'exteriorColor',
+  'trimName': 'trimDesc',  // trim.name
+  // 'Drivetrain': 'drivetrainDesc',
+  'totalPrice': 'price',  // pricing.cash.summary.items[].value where items[].type==total_vehicle_price
+  'vehicleAvailabilityDisplayStatus': 'inventoryStatus',
```
[the remark this answers is not in the document]
Comment (blamejoel) [reply app/src/manufacturers/chevroletMappings.js:24]: Whoops, artifact from using genesisMappings.js as a starting reference. Fixed.

File: api/src/blueprints/chevrolet.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+chevrolet = Blueprint(name="chevrolet", import_name=__name__)
+
+api_url = 'https://www.chevrolet.com/electric/shopping/api/drp-cp-api/p/v1/vehicles'
+
+@chevrolet.route('/api/inventory/chevrolet', methods=['GET'])
+def get_chevrolet_inventory():
+  params = {
+      'zip': request.args.get('zip'),
+      'year': request.args.get('year'),
+      'model': request.args.get('model'),
```
[the remark this answers is not in the document]
Comment (blamejoel) [reply api/src/blueprints/chevrolet.py:16]: Hm. Accessing the request.args this way (`.get(<arg>)`) should follow the [application/x-www-form-urlencoded spec](https://www.w3.org/TR/html401/interact/forms.html#h-17.13.4.1) and auto decode the `+` into a space automatically. Are you seeing an issue on a deployed staging instance? It's decoding correctly on my local dev venv, but if it's not working on the deployed I'd also be curious as to what other nuances may show up elsewhere.

File: api/src/blueprints/chevrolet.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+from flask import Blueprint, request
+
+from libs.libs import send_response, send_error_response, validate_request
+
+chevrolet = Blueprint(name="chevrolet", import_name=__name__)
+
+api_url = 'https://www.chevrolet.com/electric/shopping/api/drp-cp-api/p/v1/vehicles'
+
+@chevrolet.route('/api/inventory/chevrolet', methods=['GET'])
+def get_chevrolet_inventory():
```
[the remark this answers is not in the document]
Comment (blamejoel) [reply api/src/blueprints/chevrolet.py:12]: Yikes, that spacing was all over the place; good catch, thanks!

**Comment by blamejoel** [comment]
I looked around at the Chevrolet API and there's an endpoint for getting dealer info including dealer URLs, but you have to query it for each dealer--which could potentially turn into a lot of additional network requests on each search.

Since it's unlikely the URL for any given dealer will change all that frequently, we could cache the dealer <-> URL mapping on localStorage/vue store or another more persistent option (db). Regardless of strategy, an API call (local or external) will be needed for each new dealer in the `Set` of dealers from all returned results to allow the dealer name to be hyperlinked for Chevrolet.

Alternatively, we could make the dealer name clickable and just use a clickhandler for Chevrolet specifically that gets the URL from the API at runtime (on click) and navigates to it; the tradeoff here would be having to wait on the synchronous API call to fetch the dealer URL, _and then_ loading the URL in the new page. It's also worth noting that some browsers may not let the JavaScript "build" the URL to open in a new window like this so we'd have do a workaround by opening the new window, making the fetch request to load the resource, then writing the data to the new window.

File: api/src/blueprints/chevrolet.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ben-Chapman) [reply api/src/blueprints/chevrolet.py]: Huh, interesting, it is something with my local environment. I deployed this out to dev (dev.theevfinder.com) and it's working as expected - so ... nevermind :)

**Comment by Ben-Chapman** [comment]
It's interesting that the Chevrolet API does not return _any_ information for a dealer, unless you dill down into the detail for a specific vehicle. I like the ClickHandler idea, but in looking at the stats few people (<5%) click on the dealer link so I think it's fine as is.

I just pushed the hero imagery into your fork, and from what I can see this is good to ship.

**Pull request merged**